compiler/x/codegen/DataSnippet.cpp

```diff
@@ -80,6 +79,16 @@ TR::X86DataSnippet::addMetaDataForCodeAddress(uint8_t *cursor)
             cg()->jitAdd32BitPicToPatchOnClassRedefinition(((void *) -1), (void *) cursor, true);
             }
          }
+
+      TR_OpaqueClassBlock *clazz = getData<TR_OpaqueClassBlock *>();
+      if (clazz && cg()->comp()->compileRelocatableCode() && cg()->comp()->getOption(TR_UseSymbolValidationManager))
+         {
+         cg()->addExternalRelocation(new (cg()->trHeapMemory()) TR::ExternalRelocation(cursor,
+                                                               (uint8_t *)clazz,
+                                                               (uint8_t *)TR::SymbolType::typeClass,
+                                                               TR_SymbolFromManager,
+                                                               cg()),  __FILE__, __LINE__, getNode());
+         }
```

Comment:
Yeah that's a good idea; although before this code, we never created a DataSnippet for a class pointer under AOT - this code is now active under AOT because checkcast inlining was enabled.